fbw-a380x/src/systems/instruments/src/MFD/pages/common/InputField.tsx

```diff
@@ -0,0 +1,604 @@
+
+  private modifiedFieldValue = Subject.create<string | null>(null);
+
+  private isFocused = Subject.create(false);
+
+  private isValidating = Subject.create(false);
+
+  private alignTextSub: Subscribable<'flex-start' | 'center' | 'flex-end'> = SubscribableUtils.toSubscribable(
+    this.props.alignText ?? 'center',
+    true,
+  );
+
+  private onNewValue() {
+    // Don't update if field is being edited
+    if (this.isFocused.get() === true || this.isValidating.get() === true) {
+      return;
+    }
+
+    // Reset modifiedFieldValue
+    if (this.modifiedFieldValue.get() !== null) {
+      this.modifiedFieldValue.set(null);
+    }
+    if (this.props.value.get() !== undefined) {
+      if (this.props.canOverflow === true) {
+        // If item was overflowed, check whether overflow is still needed
+        this.overflow(!((this.props.value.get()?.toString().length ?? 0) <= this.props.dataEntryFormat.maxDigits));
+      }
+
+      if (this.props.mandatory?.get() === true) {
+        this.textInputRef.getOrDefault()?.classList.remove('mandatory');
+      }
+    }
+    this.updateDisplayElement();
+  }
+
+  private updateDisplayElement() {
+    // If input was not modified, render props' value
+    if (this.modifiedFieldValue.get() === null) {
+      if (this.props.value.get() === null || this.props.value.get() === undefined) {
+        this.populatePlaceholders();
+      } else {
+        const [formatted, leadingUnit, trailingUnit] = this.props.dataEntryFormat.format(this.props.value.get());
+        this.textInputRef.instance.innerText = formatted ?? '';
+        this.leadingUnit.set(leadingUnit ?? '');
+        this.trailingUnit.set(trailingUnit ?? '');
+      }
+    } else {
+      // Else, render modifiedFieldValue
+      const numDigits = this.props.dataEntryFormat.maxDigits;
+      if (
+        (this.modifiedFieldValue.get()?.length ?? 0) < numDigits ||
+        this.isFocused.get() === false ||
+        this.props.canOverflow === true
+      ) {
+        this.textInputRef.instance.innerText = this.modifiedFieldValue.get() ?? '';
+        this.caretRef.instance.innerText = '';
+      } else {
+        this.textInputRef.instance.innerText = this.modifiedFieldValue.get()?.slice(0, numDigits - 1) ?? '';
+        this.caretRef.instance.innerText = this.modifiedFieldValue.get()?.slice(numDigits - 1, numDigits) ?? '';
+      }
+    }
+  }
+
+  // Called when the input field changes
+  private onInput() {
+    if (
+      this.props.canOverflow === true &&
+      this.modifiedFieldValue.get()?.length === this.props.dataEntryFormat.maxDigits
+    ) {
+      this.overflow(true);
+    }
+
+    if (this.props.onInput) {
+      this.props.onInput(this.modifiedFieldValue.get() ?? '');
+    }
+  }
+
+  public overflow(overflow: boolean) {
+    if (this.topRef.getOrDefault() && this.containerRef.getOrDefault()) {
+      if (overflow === true) {
+        this.topRef.instance.style.position = 'relative';
```

Comment:
Can we do this by toggling classes instead of modifying the style property directly?

Reply:
done. Except for the width property